plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateCommandWrapper.java

```diff
@@ -415,6 +415,20 @@ private String getSourceFileDevText(Node diskNode) {
                 if (diskNodeAttribute != null) {
                     return diskNodeAttribute.getTextContent();
                 }
+
+                diskNodeAttribute = diskNodeAttributes.getNamedItem("protocol");
+
+                if (diskNodeAttribute != null) {
```

Comment:
@mike-tutkowski can `diskNodeAttribute` be blank or empty? If yes a `StringUtils.isBlank(diskNodeAttribute)` would fit in this conditional.

Additionally, Do you think that we could benefit from an elaborated check here? For instance, verify if it is a valid protocol.